cdap-app-fabric/src/test/java/co/cask/cdap/internal/profile/ProfileServiceTest.java

```diff
@@ -213,6 +213,9 @@ public void testProfileService() throws Exception {
     } catch (NotFoundException e) {
       // expected
     }
+
+    // clean up native profile
+    profileService.deleteAllProfiles(NamespaceId.SYSTEM);
```

Comment:
I see this method is documented that it should only be called by unit tests. What about when a namespace is deleted? Doesn't it need to call this too (and did we forget to implement that)?

I think delete(namespace) should be a normal method, except you are not allowed to pass NamespaceId.SYSTEM. For unit tests, we should have a package private clear() that deletes everything but the native profile. There should then be a `@After` method in this class that always clears data in between tests.

Reply:
Good catch, I forgot to implement delete profiles on namespace deletion, will add that in the next commit